src/AutorestSwift/View Models/ServiceClientViewModel.swift

```diff
@@ -42,23 +46,184 @@ struct ParameterViewModel {
     }
 }
 
+struct KeyValueViewModel {
+    let key: String
+    let value: String
+
+    init(from schema: Parameter, signatureParameters params: [ParameterViewModel]) {
+        self.key = schema.serializedName!
+        self.value = getValue(from: schema, signatureParameters: params)
+    }
+}
+
+private func getValue(from schema: Parameter, signatureParameters params: [ParameterViewModel]) -> String {
+    if let constantSchema = schema.schema as? ConstantSchema {
+        return "\"\(constantSchema.value.value)\""
+    }
+
+    // check if the name matches with one of the signature parameter
+    // if match, the value is taken from the signature parameter
+    for param in params where param.name == schema.name {
+        return param.name
+    }
+
+    return ""
+}
+
+struct RequestViewModel {
+    let path: String
+    let method: String
+    let knownMediaType: String?
+    let uri: String?
+    let mediaTypes: [String]?
+    let params: [ParameterViewModel]?
+    let objectType: String?
+
+    init(from request: Request) {
+        // load HttpRequest properties
+        if let httpRequest = (request.protocol.http as? HttpRequest) {
+            self.path = httpRequest.path
+            self.method = httpRequest.method.rawValue
+            self.uri = httpRequest.uri
+        } else {
+            self.path = ""
+            self.method = ""
+            self.uri = ""
+        }
```

Comment:
More concise suggestion:

```suggestion
    init(from request: Request) {
        // load HttpRequest properties
        let httpRequest = (request.protocol.http as? HttpRequest)
        self.path = httpRequest?.path ?? ""
        self.method = httpRequest?.method.rawValue ?? ""
        self.uri = httpRequest?.uri ?? ""
```

Reply:
Thank for the tip. Appreciated.